### ISSUE TITLE
How to pass https proxy information to ServiceURL  azure blob storage connection in Azure Java Storage SDK V10

### ISSUE THREAD

**Issue by chhajerpawan** [Azure/azure-storage-java#453]
### Which service(blob, file, queue, table) does this issue concern?

Blob Storage 
### Which version of the SDK was used?
V10

### What problem was encountered?
How to pass https proxy information to Connection

**Comment by rickle-msft** [comment]
Hi, @chhajerpawan. You should be able to use [this](https://github.com/Azure/azure-storage-java/blob/master/src/test/java/com/microsoft/azure/storage/APISpec.groovy#L186) as a guide (through line 211, and you can skip the logging bits if you want). Please let me know if you have any difficulties. 

I noticed you were on v8 in another issue and you are asking about v10 here. Our of curiosity, are you trying to use both at once and load them side by side? Or is there something in particular that's driving you to upgrade?

**Comment by chhajerpawan** [comment]
Here with V10 i am trying to simulate upload speed , so just making a sample app with upload file functionality so that i can simulate upload speed with V10

**Comment by rickle-msft** [comment]
@chhajerpawan Do you have any updates on whether the sample I pointed you to helped unblock your use of proxies?

**Comment by chhajerpawan** [comment]
Proxy is not working , without proxy i am able to upload data

**Comment by rickle-msft** [comment]
If this is still an issue for you, can you please describe what failures you are seeing with the proxy after looking at the sample code I pointed you to?

**Comment by chhajerpawan** [comment]
WARNING: An exception was thrown by com.microsoft.rest.v2.http.NettyClient$AcquisitionListener.operationComplete()
java.lang.NullPointerException
	at com.microsoft.rest.v2.http.NettyClient$AcquisitionListener.emitError(NettyClient.java:426)
	at com.microsoft.rest.v2.http.NettyClient$AcquisitionListener.operationComplete(NettyClient.java:247)
	at io.netty.util.concurrent.DefaultPromise.notifyListener0(DefaultPromise.java:511)
	at io.netty.util.concurrent.DefaultPromise.notifyListenersNow(DefaultPromise.java:485)
	at io.netty.util.concurrent.DefaultPromise.access$000(DefaultPromise.java:33)
	at io.netty.util.concurrent.DefaultPromise$1.run(DefaultPromise.java:435)
	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:163)
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:404)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:464)
	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:884)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.lang.Thread.run(Unknown Source)

**Comment by rickle-msft** [comment]
It looks like that issue may be related to [this one](https://github.com/Azure/azure-storage-java/issues/440), which we should have a fix for relatively soon. It may not resolve the whole proxy situation (it's not clear why there's an error in the first place when using a proxy), but it should be a step in the right direction and help us further diagnose the issue.

**Comment by mkcello96** [comment]
> Hi, @chhajerpawan. You should be able to use [this](https://github.com/Azure/azure-storage-java/blob/master/src/test/java/com/microsoft/azure/storage/APISpec.groovy#L186) as a guide (through line 211, and you can skip the logging bits if you want). Please let me know if you have any difficulties.
> 
> I noticed you were on v8 in another issue and you are asking about v10 here. Our of curiosity, are you trying to use both at once and load them side by side? Or is there something in particular that's driving you to upgrade?

@rickle-msft Could you provide an updated link for that guide? I'm trying to use a proxy with a ServiceURL as well.

**Comment by rickle-msft** [comment]
Here is an updated [link](https://github.com/Azure/azure-sdk-for-java/blob/master/storage/data-plane/src/test/java/com/microsoft/azure/storage/APISpec.groovy#L194)

You should know that we are pausing development on storage v11 and client-runtime v2. We have released a preview of v12 which fixes many networking-related errors frequently encountered in v11. It follows a similar pattern but has a bit of a facelift in the way of naming helpers. If you are uncomfortable working with a preview version, we are also still supporting and developing v8 which does not face this issue. If you have any questions about this versioning story, I will be happy to explain more, but for now we are directing customers away from v11.

**Comment by mkcello96** [comment]
@rickle-msft Thanks for the heads up. 
We're moving to v11 since we need the storage tier APIs as well. Would v10 not work?
If not, I'm interested in the network issues you're describing.

Thanks!

**Comment by rickle-msft** [comment]
Which apis specifically are you referring to? v8 should have support for all tiering related operations as we've continued to support that architecture in parallel with v10/11. 

v10 and v11 are both of the same architecture and rest on the same networking stack that has the problems I referred to. v8 is the older version that is much more stable and mature and is still fully supported. We also just released a preview of v12, which looks similar but not identical to v10/11 but uses a much better networking stack as it is no longer a custom implementation. The recommendation right now is either v8 or v12preview.

The networking issues were mostly seeing a lot of Connection Reset by Peer exceptions in long running workloads. Certain cases would slowly drain the connection pool and eventually cause it to be empty, which would force exceptions to bubble up and the program to lock until it was restarted.
